systemvm/debian/opt/cloud/bin/cs/CsDhcp.py

```diff
@@ -127,6 +127,9 @@ def configure_server(self):
                 listen_address.append(gateway)
             else:
                 listen_address.append(ip)
+            # Add localized "data-server" records in /etc/hosts for VPC routers
+            if self.config.is_vpc():
+                self.add_host(gateway, "%s data-server" % CsHelper.get_hostname())
```

Comment:
this can only work if we know for sure that `CsHelper.get_hostname()` gives an ip address and no hostname, are we?

Reply:
> this can only work if we know for sure that `CsHelper.get_hostname()` gives an ip address and no hostname, are we?

@DaanHoogland 
`CsHelper.get_hostname()` gives the hostname of virtual router. As you can see in screenshot, 'r-24-VM' is the hostname. So it should return hostname, not IP.

Reply:
sorry, never mind, I mis-interpreted the semantics of add_host(). no ip needed in that parameter, a (list of) hosts-name(s) is fine.